### PR TITLE
support new registry option

### PR DESCRIPTION
**Pull request by kwmonroe**
Let users specify the docker registry they wish to use when deploying
addons. This facilitates offline deployments.

### CODE REVIEW

File: get-addon-templates [hunk 1]
```diff
@@ -102,7 +102,9 @@ def add_addon(repo, source, dest, required=True, base='cluster/addons'):
 		"k8s.gcr.io/addon-resizer:1.8.1",
 		"cdkbot/addon-resizer-{{ arch }}:1.8.1"
     )
-
+    content = content.replace("cdkbot", "{{ registry|default('cdkbot') }}")
+    content = content.replace("k8s.gcr.io", "{{ registry|default('k8s.gcr.io') }}")
+    content = content.replace("nvidia", "{{ registry|default('nvidia') }}")
```
Comment (Cynerva) [inline get-addon-templates]: The content-wide replacement of `nvidia` is scary. Consider [nvidia-device-plugin.yml](https://github.com/NVIDIA/k8s-device-plugin/blob/f66ffd688edb808b2f2f93458437f1ac2c236cf7/nvidia-device-plugin.yml) - replacing `nvidia` in here will butcher the template pretty bad.

Not that the existing code in here is a shining example of anything. The same concern applies to content-wide replacements of `amd64`, etc. But the `nvidia` case seems particularly bad - I don't think we can ship this.

Doing a replace on something like `image: nvidia/` might be more acceptable, but still dirty. Probably more ideal would be to `yaml.load` the data, walk through and find the `image` fields, and only modify those.

Comment (kwmonroe) [reply get-addon-templates]: DOH! great catch @Cynerva.  I put 'nvidia' in there as an afterthought (not testing on gpu-enabled hardware).  i'll play around with the 'image: ' munging.

**Comment by kwmonroe** [comment]
@Cynerva `yaml.load_all` proved difficult to walk templates with multi docs (like [heapster-controller](https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/cluster-monitoring/standalone/heapster-controller.yaml)) because pyyaml gets confused on templated items like `{{ foo }}`.

I pushed a version that does the replacement on `image: foo/`, and to be less dirty, i did it with `re.sub` so we catch instances where the templates have wacky spacing between `image:` and `foo`. How does this sit with you?

**Comment by kwmonroe** [comment]
FYI, testing looks like this.  Deploy cdk, check the addon images:
```
root@ip-172-31-9-89:~/snap/cdk-addons/x1/addons# grep image *
heapster-controller.yaml:        image: k8s.gcr.io/heapster-amd64:v1.5.3
heapster-controller.yaml:        image: k8s.gcr.io/heapster-amd64:v1.5.3
heapster-controller.yaml:        image: cdkbot/addon-resizer-amd64:1.8.1
heapster-controller.yaml:        image: cdkbot/addon-resizer-amd64:1.8.1
influxdb-grafana-controller.yaml:      - image: k8s.gcr.io/heapster-influxdb-amd64:v1.3.3
influxdb-grafana-controller.yaml:        image: k8s.gcr.io/heapster-grafana-amd64:v4.4.3
kube-dns.yaml:        image: k8s.gcr.io/k8s-dns-kube-dns-amd64:1.14.10
kube-dns.yaml:        image: k8s.gcr.io/k8s-dns-dnsmasq-nanny-amd64:1.14.10
kube-dns.yaml:        image: k8s.gcr.io/k8s-dns-sidecar-amd64:1.14.10
kubernetes-dashboard.yaml:        image: k8s.gcr.io/kubernetes-dashboard-amd64:v1.8.3
metrics-server-deployment.yaml:        image: k8s.gcr.io/metrics-server-amd64:v0.2.1
metrics-server-deployment.yaml:        image: cdkbot/addon-resizer-amd64:1.8.1
```

Set the cdk-addons registry with `sudo snap set cdk-addons registry='wat:5000'`, re-run `cdk-addons.apply` and check again:
```
root@ip-172-31-9-89:~/snap/cdk-addons/x1/addons# grep image *
heapster-controller.yaml:        image: wat:5000/heapster-amd64:v1.5.3
heapster-controller.yaml:        image: wat:5000/heapster-amd64:v1.5.3
heapster-controller.yaml:        image: wat:5000/addon-resizer-amd64:1.8.1
heapster-controller.yaml:        image: wat:5000/addon-resizer-amd64:1.8.1
influxdb-grafana-controller.yaml:      - image: wat:5000/heapster-influxdb-amd64:v1.3.3
influxdb-grafana-controller.yaml:        image: wat:5000/heapster-grafana-amd64:v4.4.3
kube-dns.yaml:        image: wat:5000/k8s-dns-kube-dns-amd64:1.14.10
kube-dns.yaml:        image: wat:5000/k8s-dns-dnsmasq-nanny-amd64:1.14.10
kube-dns.yaml:        image: wat:5000/k8s-dns-sidecar-amd64:1.14.10
kubernetes-dashboard.yaml:        image: wat:5000/kubernetes-dashboard-amd64:v1.8.3
metrics-server-deployment.yaml:        image: wat:5000/metrics-server-amd64:v0.2.1
metrics-server-deployment.yaml:        image: wat:5000/addon-resizer-amd64:1.8.1
```

**Comment by hyperbolic2346** [comment]
LGTM

**Comment by ktsakalozos** [comment]
LGTM +1

**Comment by ktsakalozos** [comment]
We might want this pushed to 1.11 branch of cdk addons

**Comment by Cynerva** [comment]
LGTM

> We might want this pushed to 1.11 branch of cdk addons

Good catch, thanks. Merging this to master will only affect CI builds for 1.12 and onward. We probably want to cherry-pick this into `release-1.11`, and possibly `release-1.10` and `release-1.9` as well. @kwmonroe leaving you in charge of making that happen.

**Pull request merged**